### PR TITLE
Update to Rust 2018 edition

### PR DESCRIPTION
**Pull request by sdroege**

### CODE REVIEW

File: benches/http.rs [hunk 1]
```diff
@@ -4,11 +4,11 @@ extern crate cookie_factory;
 
 use std::iter::repeat;
 
-use cookie_factory::*;
+
```
Comment (Keruspe) [inline benches/http.rs]: nuke this empty line?

File: examples/http.rs [hunk 2]
```diff
@@ -1,9 +1,9 @@
 extern crate cookie_factory;
 
-use std::{str, iter::repeat};
+use std::{str};
```
Comment (Keruspe) [inline examples/http.rs]: braces look weird when there's only one import

Comment (Keruspe) [reply examples/http.rs]: We might want to just inline the std::str where it's used

File: src/combinators/mod.rs [hunk 3]
```diff
@@ -1,6 +1,6 @@
-use gen::GenError;
-use lib::std::io::Write;
-use lib::std::io;
+use crate::gen::GenError;
+use crate::lib::std::io::Write;
+use crate::lib::std::io;
```
Comment (Keruspe) [inline src/combinators/mod.rs]: maybe make this `use crate::lib::std::io::{self, Write}` ?

File: benches/http.rs (hunk 1 above)
Comment (sdroege) [reply benches/http.rs]: Done. No idea why `cargo fix` didn't remove the empty line :)

File: examples/http.rs (hunk 2 above)
Comment (sdroege) [reply examples/http.rs]: Done

File: src/combinators/mod.rs (hunk 3 above)
Comment (sdroege) [reply src/combinators/mod.rs]: Done

**Comment by sdroege** [comment]
Should be good to go then

**Pull request merged**